### ISSUE TITLE
Implement parent function of action management's do_lua_action() in operations.c

### ISSUE THREAD

**Issue by hugo-chan** [uchicago-cs/chiventure#573]
Currently, there are two layers of functions for running actions. The higher-layer resides in operations.c/.h in CLI that takes the command from the command line, conducts validation on the action and item, then passes the required arguments to call the lower-layer function in actionmanagement.c, which does the executing of the action.

In this issue, I will be writing the function in the higher-layer. Specifically, I will have to include the setup/overhead (such as validating the command, the room, etc.), extract the necessary information from the `chiventure_ctx_t` struct, and then pass it to the lower level function in actionmanagement.c, which has not been created yet.

I expect to encounter a few obstacles. One that I have identified is that this higher level function will have to pass an action struct to the lower-level function, which is currently created using the find_action() function. However, as chiventure is moving away from a table of hard-coded actions, this may not work in the future. Therefore, sub-issues may arise, and I will document my findings/solutions/stubs here.

I will know this issue is done when I have written the function, it compiles, and all the arguments needed in the lower-layer function in actionmanagement are passed correctly.

**Comment by hugo-chan** [comment]
As expected, I am encountering a few challenges. I have been playing around in the `custom-scripts/action_functions` branch, and I have successfully added a new function in `operations.c/.h` that will call a function in action management. However, the way actions are run in chiventure appears to be more abstract than I expected: 

Currently, a list of actions that are supported is hard-coded.
This list is then converted to a lookup table where each action is stored with a function (kind_action_operation() in operations.c/.h) that corresponds to the action's type
This lookup table is stored in the chiventure game (ctx) struct
The functions are called when their actions are inputted in the command line (through find_action() and then do_cmd())

Based on this implementation, there are two ways we can approach this. One, to hard code specific names of custom script functions in the supported action list and specify a new type for custom actions.  This will allow us to write a function that is separate from the other three functions. Second, to find a way to integrate with the current three kinds of functions, but this will be hard because how do we know if an action has a custom script if it doesn't have a different type number? I am a bit stumped and will think about this more.

**Comment by hugo-chan** [comment]
I have written the foundations of the parent actionmanagement function in operation.c and also provided the backbone structure of the do_lua_function() in operation.c . However, these functions currently only support type 1.

I had followed the first method described in the previous comment. I hard-coded a LUA name command and CSCRIPT type, added a parent function that runs if the action inputted has CSCRIPT, and provided the structure for the child function in operation.c that will run the action (extract the custom script and run the custom script).
I added a custom_script field in the game_action_t struct (and modified game_action_init() and game_action_new() accordingly) so that we can actually extract the script from an action in the future.

Didn't expect this issue to involve so many parts of chiventure! I have dipped around CLI, action management, and game state so far.

**Comment by hugo-chan** [comment]
Update on this:
As addressed above, I have completed writing the parent function and provided the basic structure of the `do_lua_action()` function in actionmanagement.c. However, speaking with Tammy, who was originally assigned to implement the `do_lua_action()`, we thought that would be unfeasible in the current state of Chiventure. So after consulting with a senior developer, we decided to shift our focus from supporting custom scripts for actions to game state.

In terms of the impacts on this issue, we are planning to change our direction and, therefore, the code I had written in this issue will not be used. However, it may perhaps be of use in the future. The code can be found in `custom-scripts/action_functions`, and we will be starting a new branch for game state in the future. As a result of this change in focus, this issue is closed.
